src/parsing_routines_matrixnetworks.jl

```diff
@@ -1,123 +1,123 @@
-                push!(subs, species[i])
+        for j = 1:numspecs
+            scoef = mn.substoich[i,j]
+            if (scoef > zero(scoef))
+                push!(subs, species[j])
                 push!(sstoich, scoef)
             end
 
-            pcoef = prodstoich[i,j]
-            if (pcoef > zero(pcoef)) 
-                push!(prods, species[i])
+            pcoef = mn.prodstoich[i,j]
+            if (pcoef > zero(pcoef))
+                push!(prods, species[j])
                 push!(pstoich, pcoef)
             end
         end
 
-        addreaction!(rn, Reaction(rateexprs[j], subs, prods, sstoich, pstoich))
+        addreaction!(rn, Reaction(mn.rateexprs[i], subs, prods, sstoich, pstoich))
     end
 
     ParsedReactionNetwork(rn, nothing)
 end
 
-""" 
-    For sparse matrices
-"""
-function loadrxnetwork(ft::MatrixNetwork,
-                        rateexprs::AbstractVector, 
-                        substoich::SparseMatrixCSC, 
-                        prodstoich::SparseMatrixCSC; 
-                        species::AbstractVector=Any[], 
-                        params::AbstractVector=Any[])
-
-    sz = size(substoich)
-    @assert sz == size(prodstoich)
-    numspecs = sz[1]
-    numrxs = sz[2]
+# for sparse matrices
+function loadrxnetwork(mn::MatrixNetwork{S,T,U,V,W,X}) where {S<:AbstractVector,
+        T<:SparseMatrixCSC,U<:SparseMatrixCSC{Int,Int},V<:AbstractVector, W<:AbstractVector,X <: Any}
+    sz = size(mn.substoich)
+    @assert sz == size(mn.prodstoich)
+     numrxs= sz[1]
+    numspecs = sz[2]
 
     # create the network
     rn = make_empty_network()
-    t  = ModelingToolkit.get_iv(rn)
+    t = (mn.t === nothing) ? (@variables t)[1] : mn.t
 
     # create the species if none passed in
-    isempty(species) && (species = [funcsym(:S,t,i) for i=1:numspecs])
+    species = isempty(mn.species) ? [funcsym(:S,t,i) for i = 1:numspecs] : mn.species
     foreach(s -> addspecies!(rn, s, disablechecks=true), species)
 
     # create the parameters
-    foreach(p -> addparam!(rn, p, disablechecks=true), params)
+    foreach(p -> addparam!(rn, p, disablechecks=true), mn.params)
 
     # create the reactions
-    srows = rowvals(substoich)
-    svals = nonzeros(substoich)
-    prows = rowvals(prodstoich)
-    pvals = nonzeros(prodstoich)
-    for j = 1:numrxs
-        subs    = Any[]
-        sstoich = Vector{eltype(substoich)}()
-        prods   = Any[]
-        pstoich = Vector{eltype(prodstoich)}()
-
-        for ir in nzrange(substoich, j)
-           i     = srows[ir]
-           scoef = svals[ir]
-           if scoef > zero(scoef)
-                push!(subs, species[i])
-                push!(sstoich, scoef)
-           end
+    for i = 1:numrxs
+        sstoich = mn.substoich[i,:].nzval
+        pstoich = mn.prodstoich[i,:].nzval
+
+        if isempty(sstoich) && !isempty(pstoich)
+            subs = nothing;     sstoich = nothing
+            prods = species[mn.prodstoich[i,:].nzind]
```

Comment:
Use the sparse matrix interface like the old code did (`rowvals`, `nonzeros`, `nzrange`). You shouldn't be accessing fields in sparse matrices directly.

Reply:
> Use the sparse matrix interface like the old code did (`rowvals`, `nonzeros`, `nzrange`). You shouldn't be accessing fields in sparse matrices directly.

With new Catalyst changes of changing matrices to numspecies x numreactions, i think the old code will be workign fine.. These changes i did for `sparse version` were just to accomodate my dropping of transpose. :grinning: